src/AlsaBackedMixerElement.vala

```diff
@@ -0,0 +1,105 @@
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
+ * MA 02110-1301, USA.
+ */
+
+/* AlsaBackedMixerElement.vala - Implementation of IMixerElementViewModel using ALSA API */
+
+using Alsa;
+
+namespace BrickManager {
+    public class AlsaBackedMixerElement: IMixerElementViewModel, Object {
+        private const SimpleChannelId primary_channel_id = SimpleChannelId.MONO;
+
+        private unowned MixerElement alsa_element;
+        private SimpleElementId alsa_id;
+
+        public AlsaBackedMixerElement(MixerElement element) {
+            this.alsa_element = element;
+            SimpleElementId.alloc(out alsa_id);
+            element.get_id(alsa_id);
+        }
+
+        public string name {
+            get {
+                return alsa_id.get_name();
+            }
+        }
+
+        public uint index {
+            get {
+                return alsa_id.get_index();
+            }
+        }
+
+        public int volume {
+            get {
+                long volume = 0;
+                alsa_element.get_playback_volume(primary_channel_id, out volume);
+
+                long min_volume, max_volume;
+                alsa_element.get_playback_volume_range(out min_volume, out max_volume);
+                
+                // Prevent division by zero
+                if(max_volume == min_volume)
+                    return 0;
+                
+                // Do calculations as floats so avoid odd-looking increments from truncation
+                return (int)Math.round((volume - min_volume) * 100 / (float)(max_volume - min_volume));
+            }
+            set {
+                long min_volume, max_volume;
+                alsa_element.get_playback_volume_range(out min_volume, out max_volume);
+
+                int constrained_volume = int.min(100, int.max(0, value));
+                float scaled_volume = constrained_volume * (max_volume - min_volume) / 100f + min_volume;
+                long rounded_volume = (long)Math.round(scaled_volume);
+
+                alsa_element.set_playback_volume_all(rounded_volume);
+
+                bool should_mute = rounded_volume <= min_volume;
+                if(is_muted != should_mute)
+                    set_is_muted(should_mute);
+            }
+        }
+
+        public bool can_mute {
+            get {
+                return alsa_element.has_playback_switch();
+            }
+        }
+
+        public bool is_muted {
+            get {
+                if(!can_mute)
+                    return false;
+
+                int mute_switch = 0;
+                alsa_element.get_playback_switch(primary_channel_id, out mute_switch);
+
+                return mute_switch == 0;
+            }
+        }
+
+        protected void set_is_muted(bool is_muted) {
```

Comment:
You can't add a setter to the property because the interface defines it as get-only. So, separate setters. This behavior is different from that of C# and friends.